File: src/models/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

/**
 *
 * @author choua
 */
public class Produit {
    
    private int id,quantite_prod; 
    private float prix_prod;
    private String nom_prod, description_prod, image; 
    private Categorie  categorie; 

    public Produit() {
    }

    public Produit(int id, int quantite_prod, float prix_prod, String nom_prod, String description_prod, String image, Categorie categorie) {
        this.id = id;
        this.quantite_prod = quantite_prod;
        this.prix_prod = prix_prod;
        this.nom_prod = nom_prod;
        this.description_prod = description_prod;
        this.image = image;
        this.categorie = categorie;
    }

    public Produit(int quantite_prod, float prix_prod, String nom_prod, String description_prod, String image, Categorie categorie) {
        this.quantite_prod = quantite_prod;
        this.prix_prod = prix_prod;
        this.nom_prod = nom_prod;
        this.description_prod = description_prod;
        this.image = image;
        this.categorie = categorie;
    }

    public int getId() {
        return id;
    }

    public int getQuantite_prod() {
        return quantite_prod;
    }

    public float getPrix_prod() {
        return prix_prod;
    }

    public String getNom_prod() {
        return nom_prod;
    }

    public String getDescription_prod() {
        return description_prod;
    }

    public String getImage() {
        return image;
    }

    public Categorie getCategorie() {
        return categorie;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setQuantite_prod(int quantite_prod) {
        this.quantite_prod = quantite_prod;
    }

    public void setPrix_prod(float prix_prod) {
        this.prix_prod = prix_prod;
    }

    public void setNom_prod(String nom_prod) {
        this.nom_prod = nom_prod;
    }

    public void setDescription_prod(String description_prod) {
        this.description_prod = description_prod;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setCategorie(Categorie categorie) {
        this.categorie = categorie;
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id + ", quantite_prod=" + quantite_prod + ", prix_prod=" + prix_prod + ", nom_prod=" + nom_prod + ", description_prod=" + description_prod + ", image=" + image + ", categorie=" + categorie + '}';
    }
    
    
    
}

File: Hooligans-bon_plan_web/src/Entity/Produit.php
<?php

namespace App\Entity;
use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: ProduitRepository::class)]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_prod")]
    #[Groups("produit")]
    public ?int $id = null;

    #[ORM\Column(length: 30)]
    #[Groups("produit")]
    #[Assert\NotBlank(message:'ce champ est obligatoire')]
    public ?string $nom_prod = null;


    #[ORM\Column]
    #[Groups("produit")]
    #[Assert\NotBlank(message:'ce champ est obligatoire')]
    #[Assert\Positive(message:'le prix doit etre positive')]
    public ?float $prix_prod = null;

    #[ORM\Column(length: 100)]
    #[Groups("produit")]
    #[Assert\NotBlank(message:'ce champ est obligatoire')]
   public ?string $description_prod = null;

    #[ORM\Column]
    #[Groups("produit")]
    #[Assert\NotBlank(message:'ce champ est obligatoire')]
    #[Assert\Positive(message:'la quantite doit etre positive ')]
    public ?int $quantite_prod = null;

    #[ORM\Column(length: 255)]
    #[Groups("produit")]
    public ?string $image = null;

    #[ORM\ManyToOne]
    #[Groups("produit")]
    #[ORM\JoinColumn(nullable: false,name:"id_categorie",referencedColumnName:"id_categorie")]
    public ?Categorie $categorie = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomProd(): ?string
    {
        return $this->nom_prod;
    }

    public function setNomProd(string $nom_prod): self
    {
        $this->nom_prod = $nom_prod;

        return $this;
    }

    public function getPrixProd(): ?float
    {
        return $this->prix_prod;
    }

    public function setPrixProd(float $prix_prod): self
    {
        $this->prix_prod = $prix_prod;

        return $this;
    }

    public function getDescriptionProd(): ?string
    {
        return $this->description_prod;
    }

    public function setDescriptionProd(string $description_prod): self
    {
        $this->description_prod = $description_prod;

        return $this;
    }

    public function getQuantiteProd(): ?int
    {
        return $this->quantite_prod;
    }

    public function setQuantiteProd(int $quantite_prod): self
    {
        $this->quantite_prod = $quantite_prod;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }
}
